connect/src/com/telenor/connect/headerenrichment/HeLogic.java

```diff
@@ -0,0 +1,210 @@
+package com.telenor.connect.headerenrichment;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.net.NetworkCapabilities;
+import android.net.NetworkInfo;
+import android.net.NetworkRequest;
+import android.os.Build;
+
+import com.telenor.connect.ConnectSdk;
+import com.telenor.connect.utils.ConnectUrlHelper;
+
+import java.util.Date;
+import java.util.Map;
+
+public class HeLogic {
+    private static final long HE_TOKEN_TIMEOUT_MILLISECONDS = 10_000;
+    private static final boolean cantDirectNetworkTraffic = Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP;
+    private static boolean heTokenSuccess = true;
+    private static HeTokenCallback heTokenCallback;
+    private static boolean isHeTokenRequestOngoing;
+    private static HeTokenResponse heTokenResponse;
+    private static ConnectivityManager connectivityManager;
+    private static volatile Network cellularNetwork;
+    private static volatile Network defaultNetwork;
+
+    public static void initializeNetworks(Context context, boolean useStaging) {
+        connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+        if (connectivityManager != null && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            initializeCellularNetwork(useStaging);
+            initializeDefaultNetwork();
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    private static void initializeCellularNetwork(final boolean useStaging) {
+        NetworkRequest networkRequest = new NetworkRequest.Builder()
+                .addCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
+                .addTransportType(NetworkCapabilities.TRANSPORT_CELLULAR)
+                .build();
+        try {
+            connectivityManager.requestNetwork(
+                    networkRequest,
+                    new ConnectivityManager.NetworkCallback() {
+                        @Override
+                        public void onAvailable(Network network) {
+                            cellularNetwork = network;
+                            boolean noSignedInUser = ConnectSdk.getAccessToken() == null;
+                            if (noSignedInUser) {
+                                initializeHeaderEnrichment(useStaging, ConnectSdk.getLogSessionId());
+                            }
+                        }
+                    }
+            );
+        } catch (SecurityException e) {
```

Comment:
No need for logging here?

Reply:
Sorry, it's very non-obvious, but this is moved code from https://github.com/telenordigital/connect-android-sdk/blob/master/connect/src/com/telenor/connect/ConnectSdk.java#L573, and not new code.

I found the commit where it was added: 99257872c0a6d0d23f6320171af87a0ee5bdb948. So apparently it's a known bug for 6.0 and something that we wouldn't otherwise expect...
I guess we could add a comment in the code or something.